src/inferencerequest.cpp

```diff
@@ -104,10 +104,14 @@ Status InferenceRequest::getBatchSize(size_t& batchSize, size_t batchSizeIndex)
     batchSize = shape[batchSizeIndex];
     return StatusCode::OK;
 }
+
+// Assuming the request is already validated, therefore no need to check for negative values or zeros
 std::map<std::string, shape_t> InferenceRequest::getRequestShapes() const {
     std::map<std::string, shape_t> result;
```

Comment:
Shouldn't we have here signed_shape_t as well?

Reply:
We have `getRequestShapes` implementation for all 3 frontends - tfs/kserve/c-api. This implementation is to be used after request validation so we are ensured that there are no `0` or negative dimensions in request.

All 3 implementations contain this comment:
`// Assuming the request is already validated, therefore no need to check for negative values or zeros`